app/src/util.cc

```diff
@@ -342,5 +342,15 @@ std::vector<std::string> SplitString(const std::string& s,
   return split_parts;
 }
 
+std::string CreateApiIdentifier(const char* api_id, void* object) {
```

Comment:
Can you add a unit test for this?

Reply:
Done, and it even found a problem with the string math.